src/utilities/didUpgrade/didUpgrade.ts

```diff
@@ -25,33 +33,51 @@ import { makeFakeIdentityCrypto } from '../makeFakeIdentityCrypto/makeFakeIdenti
 
 interface DidTransaction {
   extrinsic: SubmittableExtrinsic;
-  did: DidUri;
+  did: Did;
 }
 
 export async function getTransaction(
   lightDidDocument: DidDocument,
+  keysToAdd: {
+    authentication: NewDidVerificationKey;
+    keyAgreement: NewDidEncryptionKey;
+  },
   keypair: KiltKeyringPair,
-  sign: SignExtrinsicCallback,
+  signers: SignerInterface[],
   submitter?: KiltAddress,
 ): Promise<DidTransaction> {
-  const { fullDid: did } = parseDidUri(lightDidDocument.uri);
+  const { fullDid: did } = parseDidUri(lightDidDocument.id);
 
-  const extrinsic = await Did.getStoreTx(
-    lightDidDocument,
+  await DidResolver.dereference(lightDidDocument.id, {});
```

Comment:
The intention of calling this is unclear, should it be explained in a comment?

Reply:
I agree, can't see how this would do anything? If you wanted to check for errors, you'd have to look at the dereferenceMetadata.error property

Reply:
Not sure where this line came from, obviously doesn't belong here 😅